### PR TITLE
Theme options, save theme preference and oscdark enhancements

### LINKED ISSUE #91: Dark Mode adjustments
I've been working with some adjustments in the dark theme of the downloader. I've done some slight color adjustments to improve readability (for example links are very hard to read on current dark mode). Only slight adjustments on some niche components, nothing too extreme since I already like the palette a lot.  

I'd also like to add the ability to switch back to the default theme through the combo box. It annoys me to have to restart for light theme, and I guess it would be helpful if someday theme selection moves out of the 'experimental' menus.

I am already working on this. Would you be interested in a future pull request once I'm sure it's ready?

### PR DESCRIPTION
**Pull request by pavlidvg**
Adds support for switching back to the System Default after enabling dark mode, as well as saving the user preference for the next startup.

2 small changes were also made to the dark theme, for improved readability. Links are now a light blue shade (similar to discord or JetBrains links on dark mode). The 'Download' dialog is now correctly dark. 
after
before

A few new code snippets are missing comments. It is because I am already working on a pull request that slightly changes them and adds proper documentation.

### CODE REVIEW

File: settings [hunk 1]
```diff
@@ -0,0 +1 @@
+{"theme": "Default"}
```
Comment (dhtdht020) [inline settings:1]: Please use the proper API for settings. (gui_helpers.settings), [documentation](https://doc.qt.io/qtforpython/PySide6/QtCore/QSettings.html) is available here. It is also used by the DownloadDialog and SendDialog if you'd like to see how to use it.

Comment (dhtdht020) [reply settings:1]: Once this is resolved, the function from utils.py can probably be removed too.

File: xosc_dl.py [hunk 2]
```diff
@@ -144,7 +145,13 @@ def __init__(self, test_mode=False):
...
 
     def about_dialog(self):
-        QMessageBox.about(self, f"About OSCDL",
+        QMessageBox.about(self, f"About OSCDL","""<style>        
+
+        a{  
+            color:#0F8EC2;
+            text-decoration: none;
+        }        
+    </style>"""
```
Comment (dhtdht020) [inline xosc_dl.py:154]: The styling here should probably be a part of the stylesheet.

Comment (pavlidvg) [reply xosc_dl.py:154]: That was my original intention, I am just not sure it's doable

I did not find any ways to edit a QMessageBox html text through the stylesheet. I even found some forum posts specifically claiming you can't change link colors in qt MessageBoxes, so any solution must be hard-coded. I'll look into it further.

Plus, I haven't found any other links anywhere one the app, so it's not like we will be saving any time by placing the styling on the stylesheet (although I get it on principle)

File: settings (hunk 1 above)
Comment (pavlidvg) [reply settings]: yup. To be frank, I didn't have any experience working with this but it's much more elegant than what I proposed. Should be fixed in the newest commit (also yes, the utils function was not needed after all)

File: xosc_dl.py (hunk 2 above)
Comment (pavlidvg) [reply xosc_dl.py]: **Update:** I am currently omitting this from the changes as you can see in the newest commit. I'll try to create a solution that is theme specific, but I have read from several sources that QLabel and QMessageBox links are not customizable through stylesheets. Would you prefer an app-wide solution  (people online prefer customizing the QPalette), or just keep the original dark blue and somewhat hinder readability on one page for the dark theme?

**Review by techydude0713 — COMMENTED** [review]
Great work. I have a few things to comment.

File: xosc_dl.py [hunk 3]
```diff
@@ -936,14 +941,108 @@ def return_to_all_apps_btn(self):
...
+
+        if not file_chosen[0]:  # in case no file was selected
+            return
+        # filename is a tuple. filename[0] is the actual file name
+
+        absolute = os.path.dirname(__file__)  # absolute file path of the app
+        filename = os.path.basename(file_chosen[0])
+
+        # copy the file over to assets/themes
+        with open(file_chosen[0],"r") as src,open(os.path.join(absolute,"assets/themes/",filename),"w") as dest:
```
[the remark this answers is not in the document]
Comment (pavlidvg) [reply xosc_dl.py:1036]: The others where a mistake on my part, but I 'm also not sure if the absolute path returned by the dialog isn't enough here. Dev works fine  
To be honest I haven't used pyinstaller in years. I can test it myself if you have build instructions for the project.

[the remark this answers is not in the document]
Comment (dhtdht020) [reply xosc_dl.py]: I am not sure if there's a need for joining the absolute path at all here. resource_path() would work for reaching the relative path in the usual manner for both normal use and frozen use.

File: xosc_dl.py (hunk 2 above)
Comment (dhtdht020) [reply xosc_dl.py]: Might have to keep the dark blue here for now.
There's another hyperlink in the program in the main window, you can find it by clicking on the icon next to the author name when selecting an app.

**Comment by pavlidvg** [comment]
New commits should correctly implement theme file loading on dev and frozen, using resource_path correctly. Also, @dhtdht020 was right, the program path join was unnecessary, I previously misunderstood the resource_path function

**Comment by dhtdht020** [comment]
Hello, I am once again sorry for the wait. I have made the decision to remove themes altogether and use the generic dark theme implementation provided by Qt. Your effort is highly appreciated.

**Pull request closed without merging**